src/master-playlist-controller.js

```diff
@@ -833,9 +831,9 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return;
     }
 
-    const isFinalRendition =
-      this.masterPlaylistLoader_.master.playlists.filter(isEnabled).length === 1;
     const playlists = this.masterPlaylistLoader_.master.playlists;
+    const enabledPlaylists = playlists.filter(isEnabled);
+    const isFinalRendition = enabledPlaylists.length === 1 && enabledPlaylists[0] === currentPlaylist;
```

Comment:
This function was probably broken for two playlist situations where one playlist gets blacklisted and the other doesn't. This is because we did not check if the last enabled playlist is the playlist we are about to blacklist. Example

You have two playlists `a` and `b` . `a` is enabled, now we are going to blacklist `a` and enable `b`. `enabledPlaylists` will contain only `b`. This means that `a` which is being blacklisted is **NOT** the final rendition.

Reply:
yeah

Reply:
Is it possible for us to run into this case?